tests/unit/test_dimension_utilities.py

```diff
@@ -304,6 +311,141 @@ def test_prefetch_dimension_variables(self, mock_get_opendap_nc4):
+
+        with self.subTest('Variable has no cell alignment and has bounds'):
+            self.assertFalse(needs_bounds(varinfo_bounds.get_variable('/variable_no_edge_has_bnds')))
+
+        with self.subTest('Variable has cell alignment and no bounds'):
+            self.assertTrue(needs_bounds(varinfo_bounds.get_variable('/variable_edge_no_bnds')))
+
+        with self.subTest('Variable has no cell alignment and no bounds'):
+            self.assertFalse(needs_bounds(varinfo_bounds.get_variable('/variable_no_edge_no_bnds')))
+
+
+    def test_get_bounds_array(self):
+        """ Ensure that the expected bounds array is created given
+            the input dimension variable values.
+
+        """
+        prefetch_dataset = Dataset('tests/data/ATL16_prefetch.nc4', 'r')
+        dimension_path = '/npolar_grid_lat'
+
+        expected_bounds_array = np.array([[90.0, 89.0], [89.0, 88.0], [88.0, 87.0], [87.0, 86.0],
+                                 [86.0, 85.0], [85.0, 84.0], [84.0, 83.0], [83.0, 82.0],
+                                 [82.0, 81.0], [81.0, 80.0], [80.0, 79.0], [79.0, 78.0],
+                                 [78.0, 77.0], [77.0, 76.0], [76.0, 75.0], [75.0, 74.0],
+                                 [74.0, 73.0], [73.0, 72.0], [72.0, 71.0], [71.0, 70.0],
+                                 [70.0, 69.0], [69.0, 68.0], [68.0, 67.0], [67.0, 66.0],
+                                 [66.0, 65.0], [65.0, 64.0], [64.0, 63.0], [63.0, 62.0],
+                                 [62.0, 61.0], [61.0, 60.0]])
+
+        assert_array_equal(get_bounds_array(prefetch_dataset, dimension_path),
+                         expected_bounds_array)
+
+
+    def test_write_bounds(self):
+        """ Ensure that bounds data array is written to the dimension
+            dataset, both when the dimension variable is in the root group
+            and in a nested group.
+
+        """
+        varinfo_prefetch = VarInfoFromDmr('tests/data/ATL16_prefetch_group.dmr')
+        prefetch_dataset = Dataset('tests/data/ATL16_prefetch_group.nc4', 'r+')
+
+        # Expected variable contents in file.
+        expected_bounds_data = np.array([
+            [90.0, 89.0], [89.0, 88.0], [88.0, 87.0], [87.0, 86.0],
+            [86.0, 85.0], [85.0, 84.0], [84.0, 83.0], [83.0, 82.0],
+            [82.0, 81.0], [81.0, 80.0], [80.0, 79.0], [79.0, 78.0],
+            [78.0, 77.0], [77.0, 76.0], [76.0, 75.0], [75.0, 74.0],
+            [74.0, 73.0], [73.0, 72.0], [72.0, 71.0], [71.0, 70.0],
+            [70.0, 69.0], [69.0, 68.0], [68.0, 67.0], [67.0, 66.0],
+            [66.0, 65.0], [65.0, 64.0], [64.0, 63.0], [63.0, 62.0],
+            [62.0, 61.0], [61.0, 60.0]])
+
+        with self.subTest('Dimension variable is in the root group'):
+            root_variable_full_path = '/npolar_grid_lat'
+            root_varinfo_variable = varinfo_prefetch.get_variable(root_variable_full_path)
+            root_variable_name = str(PurePosixPath(root_variable_full_path).name)
```

Comment:
Opinion ahoy: This feels a little complicated. I think with tests that you know a string value up front it's sometimes better to just explicitly spell them out, rather than constructing them. (It makes the tests more readable later on)

Reply:
Similar in the nested group subtest.